### ISSUE TITLE
Feature request: add input support for PostgreSQL domain over a composite type

### ISSUE THREAD

**Issue by jcgsville** [graphile/graphile-engine#600]
I'm submitting a bug report

PostGraphile version: v4.5.5

Minimal SQL file that can be loaded into a clean database:

```sql
drop schema if exists my_schema cascade;
drop domain if exists two_ints_constrained;
drop type if exists two_ints;

create schema my_schema;

create type two_ints as (
	one		integer,
	two     integer
);

create domain two_ints_constrained as two_ints NOT NULL;

create table my_schema.foo (
	f 		two_ints_constrained,
	g		integer,
	UNIQUE (f, g)
);
```

Steps to reproduce:

Run `postgraphile -c postgres://postgres@localhost:5432/my_db -s my_schema`

Current behavior:
It fails to build the schema with the following error

```
 serious error occurred when building the initial schema. Exiting because `retryOnInitFail` is not set. Error details:

Error: Could not find input type for key 'f' on type 'Foo'
    at args.keysIncludingMeta.reduce (/path/to/postgraphile/node_modules/graphile-build-pg/node8plus/plugins/PgRowByUniqueConstraint.js:108:25)
```

Expected behavior:
It behaves the same as if the the table definition does not use the domain, and just the composite type. An example is the following: (this works no problem)

```sql
create table my_schema.foo (
	f 		two_ints,
	g		integer,
	UNIQUE (f, g)
);
```

**Comment by benjie** [comment]
We need a block like this:

https://github.com/graphile/graphile-engine/blob/a9746f1d444fa6a39ab18cd983776cd68792c83e/packages/graphile-build-pg/src/plugins/PgTypesPlugin.js#L765-L797

But for input types. PR welcome :+1:

**Comment by ben-pr-p** [comment]
Should this be closed with #615 merged?
